src/main/java/org/zalando/nakadi/security/ClientResolver.java

```diff
@@ -42,7 +48,8 @@ public Client resolveArgument(final MethodParameter parameter,
                                   final ModelAndViewContainer mavContainer,
                                   final NativeWebRequest request,
                                   final WebDataBinderFactory binderFactory) throws Exception {
-        final Optional<String> clientId = Optional.ofNullable(request.getUserPrincipal()).map(Principal::getName);
+        final Optional<String> clientId = Optional.ofNullable(authorizationService.getSubject().isPresent() ?
```

Comment:
If it always has a value it's no longer an `Optional`, right?

Reply:
Right, I'll remove `Optional`